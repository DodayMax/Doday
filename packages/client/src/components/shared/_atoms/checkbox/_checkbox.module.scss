@import '@styles/_config.scss';

.checkbox-container {
  position: relative;
  height: 3.6rem;
}

.color-marker {
  position: absolute;
  right: 0;
  width: 0.8rem;
  height: 0.8rem;
  border-radius: 100%;
  border: 1px solid $dark-bg;
}
