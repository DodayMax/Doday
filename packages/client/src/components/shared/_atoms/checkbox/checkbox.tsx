import * as React from 'react';
import { Icons } from '@shared';
import { ActivityType } from '@root/lib/common-interfaces';
import { activityIconByType } from '@root/tools/activity/components/builders/activity-builder';
import { createStyles, withStyles, WithStyles, Theme } from '@material-ui/core';

interface CheckboxProps {
  checked?: boolean;
  colorMarker?: string;
  onClick?: (e: React.MouseEvent<HTMLDivElement>) => void;
  activityType: ActivityType;
}

const css = (theme: Theme) =>
  createStyles({
    checkboxContainer: {
      position: 'relative',
      height: '3.6rem',
    },
    colorMarker: {
      position: 'absolute',
      left: 0,
      width: '0.8rem',
      height: '0.8rem',
      borderRadius: '100%',
      border: `1px solid ${theme.palette.grey[900]}`,
    },
  });

export const Checkbox = withStyles(css)(
  ({
    checked = false,
    onClick,
    colorMarker,
    classes,
  }: CheckboxProps & WithStyles) => {
    return (
      <div
        className={classes.checkboxContainer}
        role="checkbox"
        onClick={onClick}
      >
        {colorMarker && (
          <div
            className={classes.colorMarker}
            style={{ backgroundColor: colorMarker }}
          />
        )}
        <Icons.Checkbox width={30} height={30} checked={checked} />
      </div>
    );
  }
);
