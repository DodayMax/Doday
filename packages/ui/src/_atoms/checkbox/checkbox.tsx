import * as React from 'react';
import { createStyles, withStyles, WithStyles, Theme } from '@material-ui/core';
import { Icons } from '../..';

interface CheckboxProps {
  checked?: boolean;
  colorMarker?: string;
  onClick?: (e: React.MouseEvent<HTMLDivElement>) => void;
}

const css = (theme: Theme) =>
  createStyles({
    checkboxContainer: {
      position: 'relative',
      height: '3rem',
    },
    colorMarker: {
      position: 'absolute',
      left: 0,
      width: '0.8rem',
      height: '0.8rem',
      borderRadius: '100%',
      border: `1px solid ${theme.palette.grey[900]}`,
    },
  });

export const Checkbox = withStyles(css)(
  ({
    checked = false,
    onClick,
    colorMarker,
    classes,
  }: CheckboxProps & WithStyles) => {
    return (
      <div
        className={classes.checkboxContainer}
        role="checkbox"
        onClick={onClick}
      >
        {colorMarker && (
          <div
            className={classes.colorMarker}
            style={{ backgroundColor: colorMarker }}
          />
        )}
        <Icons.Checkbox width={30} height={30} checked={checked} />
      </div>
    );
  }
);
