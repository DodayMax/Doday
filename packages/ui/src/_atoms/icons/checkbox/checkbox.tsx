import * as React from 'react';
import { IconProps } from '../names';
import { ActivityType } from '@doday/lib';

interface CheckboxProps extends IconProps {
  checked?: boolean;
  activityType?: ActivityType;
}

export const Checkbox = ({
  width = 20,
  height = 20,
  color = '#000',
  checked = false,
  activityType = 'do',
}: CheckboxProps) => {
  if (!checked) {
    return (
      <svg
        width={width}
        height={height}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g fill="none" fillRule="evenodd">
          <g
            strokeWidth=".534"
            transform="translate(1.122 1.66)"
            stroke={color}
          >
            <rect
              fill="#FFF"
              x=".267"
              y=".267"
              width="20.828"
              height="20.828"
              rx="3.204"
            />
            <rect
              fill={color}
              x="2.136"
              y="2.136"
              width="17.09"
              height="17.09"
              rx="1.602"
            />
          </g>
        </g>
      </svg>
    );
  } else {
    return (
      <svg
        width={width}
        height={height}
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g fill="none" fillRule="evenodd">
          <g strokeWidth=".5" transform="translate(1.122 1.66)" stroke={'#000'}>
            <rect
              fill="#FFF"
              x=".267"
              y=".267"
              width="20.828"
              height="20.828"
              rx="3.204"
            />
            <rect
              fill={color}
              x="2.136"
              y="2.136"
              width="17.09"
              height="17.09"
              rx="1.602"
            />
          </g>
          <path
            d="M14.238 22.508a.46.46 0 0 1-.066.09.474.474 0 0 1-.609.112L.453 14.956a.5.5 0 0 1-.175-.67l4.314-7.657a.474.474 0 0 1 .654-.18l4.632 2.74L14.79.475a.474.474 0 0 1 .654-.18l8.067 4.771a.5.5 0 0 1 .176.67l-9.45 16.772z"
            stroke={'#000'}
            strokeWidth=".4"
            fill="#FFF"
          />
          <path
            d="M23.096 5.816l-9.034 16.272a.346.346 0 0 1-.21.168.338.338 0 0 1-.263-.036l-7.503-4.508a.369.369 0 0 1-.126-.493L14.993.95a.346.346 0 0 1 .21-.168.338.338 0 0 1 .263.035l7.503 4.506c.166.101.222.32.127.492z"
            fill={'#000'}
          />
          <path
            d="M5.365 7.079l12.427 7.323c.08.047.137.124.161.214a.361.361 0 0 1-.034.269l-3.997 7.066a.343.343 0 0 1-.473.129L1.021 14.757a.361.361 0 0 1-.127-.483L4.89 7.208a.343.343 0 0 1 .474-.13z"
            fill={'#000'}
          />
          <path
            d="M10.747 10.296a.297.297 0 0 1-.102-.405l4.533-7.87a.304.304 0 0 1 .402-.115l6.593 3.78c.14.08.184.264.103.406l-4.533 7.869a.304.304 0 0 1-.403.115l-6.593-3.78z"
            fill="#F5CE24"
          />
          <path
            d="M8.767 14.148c-.788 1.398-2.572 1.863-3.97 1.036l-1.869-1.109-.062-.037 2.86-5.076.016.01 1.916 1.132c1.398.828 1.9 2.635 1.109 4.044zm-1.173-.694a1.62 1.62 0 0 0 .151-1.212 1.646 1.646 0 0 0-.752-.976l-.703-.416-1.545 2.738.71.421c.36.217.789.28 1.194.175.4-.102.74-.367.945-.733v.003zM14.84 17.831c-.815 1.445-2.648 1.923-4.094 1.067-1.445-.856-1.958-2.72-1.144-4.168.814-1.448 2.647-1.923 4.093-1.067 1.446.856 1.958 2.724 1.144 4.168zm-1.095-.647c.23-.406.288-.89.162-1.341-.268-.95-1.22-1.514-2.158-1.277a1.696 1.696 0 0 0-1.054.814 1.77 1.77 0 0 0-.163 1.342c.268.95 1.22 1.513 2.157 1.276a1.687 1.687 0 0 0 1.056-.814z"
            fill="#FAFAFA"
          />
          <path
            d="M15.849 9.207c.121.07.234.157.334.257a1.602 1.602 0 0 1 .403.666 1.409 1.409 0 0 1-.12 1.107l-.454.806a.11.11 0 0 1-.044.045.077.077 0 0 1-.07-.013l-2.49-1.473a.111.111 0 0 1-.049-.054.08.08 0 0 1 .009-.073l.455-.807a1.334 1.334 0 0 1 .532-.522 1.43 1.43 0 0 1 .725-.17c.27.011.535.09.769.23zm-1.532 1.09l1.37.811.078-.138.039-.072.034-.073a.713.713 0 0 0 .064-.268.76.76 0 0 0-.176-.517.868.868 0 0 0-.235-.199.839.839 0 0 0-.304-.111.775.775 0 0 0-.296.011.712.712 0 0 0-.453.346l-.12.21zM15.002 7.295a.1.1 0 0 1 .006-.104.092.092 0 0 1 .063-.046.297.297 0 0 1 .085-.006c.025.002.05.005.074.01a.052.052 0 0 0 .036 0l3.123.476a.07.07 0 0 0 .018.007.06.06 0 0 1 .046.05.061.061 0 0 1-.017.052l-.29.527a.1.1 0 0 1-.12.052l-.273-.062-.478.848.19.211c.036.034.043.09.017.132l-.296.524a.06.06 0 0 1-.034.04.058.058 0 0 1-.066-.015.071.071 0 0 0-.015-.013L15.104 7.44a.747.747 0 0 1-.053-.073 3.095 3.095 0 0 0-.049-.072zm1.84 1.323l.29-.51-.944-.218.655.728zM17.482 5.82l-.84-1.026a.292.292 0 0 1-.023-.035.06.06 0 0 1-.007-.021.036.036 0 0 1 0-.016.132.132 0 0 0 .006-.02l.355-.631a.022.022 0 0 1 .014-.014.022.022 0 0 1 .02.003.072.072 0 0 1 .019.02l.02.03 1.2 1.745.852.505a.082.082 0 0 1 .05.052.085.085 0 0 1-.009.073l-.277.491a.071.071 0 0 1-.051.038.11.11 0 0 1-.07-.018l-.858-.51-2.063-.188h-.033a.056.056 0 0 1-.022-.007.042.042 0 0 1-.016-.016.02.02 0 0 1 0-.021l.355-.63a.125.125 0 0 0 .015-.016.035.035 0 0 1 .013-.01.058.058 0 0 1 .022-.004h.04l1.288.226z"
            fill={'#000'}
          />
        </g>
      </svg>
    );
  }
};
