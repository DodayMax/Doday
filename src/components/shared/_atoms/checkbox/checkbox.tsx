import * as React from 'react';
import { Icons } from '@components';

const styles = require('./_checkbox.module.scss');

interface CheckboxProps {
  checked?: boolean;
  onClick: () => void; 
}

export const Checkbox = ({ checked = false, onClick }: CheckboxProps) => {
  return (
    <div role="checkbox" onClick={onClick}>
      <Icons.Checkbox checked={checked} />
    </div>
  );
}