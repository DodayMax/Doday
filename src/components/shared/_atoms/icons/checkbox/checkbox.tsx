import * as React from 'react';

interface CheckboxProps {
  checked?: boolean;
}

export const Checkbox = ({ checked = false }: CheckboxProps) => {
  if (!checked) {
    return <svg width="20" height="19" xmlns="http://www.w3.org/2000/svg"><g strokeWidth=".5" transform="translate(2.52 1.965)" stroke="#494949" fill="none" fillRule="evenodd"><rect fill="#FFF" x=".25" y=".25" width="14.967" height="14.967" rx="2.32"/><rect fill="#494949" x="1.547" y="1.547" width="12.374" height="12.374" rx="1.16"/></g></svg>;
  } else {
    return <svg width="20" height="19" xmlns="http://www.w3.org/2000/svg"><g fill="none" fillRule="evenodd"><g strokeWidth=".5" transform="translate(2.52 1.965)" stroke="#494949"><rect fill="#FFF" x=".25" y=".25" width="14.967" height="14.967" rx="2.32"/><rect fill="#494949" x="1.547" y="1.547" width="12.374" height="12.374" rx="1.16"/></g><path d="M11.68 17.773a.346.346 0 0 1-.05.068.364.364 0 0 1-.464.084l-9.984-5.827a.373.373 0 0 1-.134-.504l3.286-5.755a.364.364 0 0 1 .498-.136l3.527 2.06 3.742-6.55a.364.364 0 0 1 .498-.135l6.143 3.587a.372.372 0 0 1 .134.502L11.68 17.773z" stroke="#494949" strokeWidth=".3" fill="#FFF"/><path d="M18.425 5.228l-6.88 12.23a.263.263 0 0 1-.36.1l-5.713-3.39a.275.275 0 0 1-.097-.37l6.88-12.227a.263.263 0 0 1 .36-.1l5.713 3.387c.127.076.17.24.097.37z" fill="#222"/><path d="M4.922 6.177l9.464 5.504a.265.265 0 0 1 .097.363l-3.044 5.31a.263.263 0 0 1-.36.098l-9.465-5.504a.265.265 0 0 1-.096-.363l3.043-5.31a.263.263 0 0 1 .361-.098z" fill="#222"/><path d="M9.021 8.595a.221.221 0 0 1-.078-.305l3.452-5.914a.234.234 0 0 1 .307-.086l5.02 2.84c.107.06.14.199.079.305l-3.452 5.914a.234.234 0 0 1-.307.087l-5.02-2.841z" fill="#F5CE24"/><path d="M7.513 11.49c-.6 1.05-1.958 1.4-3.024.779l-1.422-.833-.048-.028 2.179-3.816.011.007 1.46.852c1.064.622 1.447 1.98.844 3.04zm-.893-.521c.158-.276.199-.604.115-.911a1.24 1.24 0 0 0-.573-.734l-.535-.313L4.45 11.07l.54.317c.274.163.6.21.91.132.303-.077.563-.276.72-.551v.002zM12.138 14.258c-.62 1.087-2.016 1.446-3.117.802-1.101-.643-1.492-2.044-.872-3.132.62-1.089 2.016-1.446 3.117-.802 1.101.643 1.492 2.047.872 3.132zm-.834-.486c.175-.305.22-.67.123-1.008a1.367 1.367 0 0 0-1.642-.96 1.29 1.29 0 0 0-.804.612c-.174.305-.22.67-.123 1.009.204.714.929 1.137 1.642.96.34-.085.63-.307.805-.613h-.001z" fill="#FAFAFA"/><path d="M12.907 7.776a1.284 1.284 0 0 1 .442.429 1.161 1.161 0 0 1 .165.546c.006.193-.042.384-.138.552l-.345.605a.083.083 0 0 1-.034.034.059.059 0 0 1-.054-.01l-1.895-1.107a.084.084 0 0 1-.038-.04.06.06 0 0 1 .007-.055l.346-.607a1.004 1.004 0 0 1 .406-.392 1.1 1.1 0 0 1 .552-.128c.206.008.407.068.586.173zm-1.167.82l1.043.61.06-.104.029-.055c.01-.018.018-.037.026-.054a.564.564 0 0 0-.085-.59.659.659 0 0 0-.179-.15.644.644 0 0 0-.232-.084.598.598 0 0 0-.225.01.541.541 0 0 0-.345.259l-.092.158zM12.262 6.34a.074.074 0 0 1 .004-.078.07.07 0 0 1 .048-.035.23.23 0 0 1 .064-.005c.02.002.038.004.057.008a.04.04 0 0 0 .027 0l2.379.358.014.005a.045.045 0 0 1 .022.077l-.221.396a.077.077 0 0 1-.091.039l-.209-.047-.363.638.144.158a.079.079 0 0 1 .013.099l-.225.394a.045.045 0 0 1-.026.03.044.044 0 0 1-.04-.003.047.047 0 0 1-.01-.008.054.054 0 0 0-.012-.01L12.339 6.45a.562.562 0 0 1-.04-.055 2.324 2.324 0 0 0-.037-.055zm1.401.994l.22-.383-.719-.164.5.547zM14.15 5.23l-.64-.77a.219.219 0 0 1-.017-.026.045.045 0 0 1-.005-.017v-.012a.098.098 0 0 0 .004-.015l.27-.474a.017.017 0 0 1 .026-.008c.006.004.01.01.015.015l.015.023.914 1.312.649.379c.018.006.032.02.038.04a.063.063 0 0 1-.007.054l-.21.369a.054.054 0 0 1-.04.028.084.084 0 0 1-.054-.013l-.652-.383-1.572-.142h-.024a.043.043 0 0 1-.018-.005.032.032 0 0 1-.012-.012.015.015 0 0 1 0-.016l.27-.474a.095.095 0 0 0 .012-.011.027.027 0 0 1 .01-.008.044.044 0 0 1 .016-.003h.032l.98.17z" fill="#000"/></g></svg>;
  }
};