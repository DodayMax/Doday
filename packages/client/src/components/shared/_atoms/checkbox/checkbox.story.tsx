import * as React from 'react';

import { storiesOf } from '@storybook/react';
import { Checkbox } from './checkbox';

storiesOf('Checkbox', module)
  .add('default', () => <Checkbox onClick={() => {}} />)
  .add('checked', () => <Checkbox checked onClick={() => {}} />);
