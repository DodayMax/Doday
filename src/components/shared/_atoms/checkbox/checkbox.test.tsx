import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { Checkbox } from './checkbox';

describe('Checkbox', () => {
  it('renders correctly without props', () => {
    const checkbox = shallow(<Checkbox onClick={() => {}} />);

    expect(checkbox).toMatchSnapshot();
  });

  it('renders correctly checked', () => {
    const checkbox = shallow(<Checkbox checked onClick={() => {}} />);

    expect(checkbox).toMatchSnapshot();
  });

  it('click trigger handler', () => {
    const onClick = jest.fn();
    const checkbox = shallow(<Checkbox onClick={onClick} />);

    checkbox.simulate('click');
    expect(onClick.mock.calls.length).toBe(1);
  });
});
