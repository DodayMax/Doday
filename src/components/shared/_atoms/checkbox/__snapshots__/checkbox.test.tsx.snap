// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly checked 1`] = `ShallowWrapper {}`;

exports[`Checkbox renders correctly without props 1`] = `ShallowWrapper {}`;
